src/io/network/socket.hpp

```diff
@@ -42,7 +42,7 @@ class Socket {
   /**
    * Shutdown the socket if it is open.
    */
-  void Shutdown();
+  void Shutdown() const;
```

Comment:
Even though technically these functions could be const, in my opinion they shouldn't. The class represents a socket object (by wrapping some pointers), and these functions are modifying the abstract state of the object. Currently these function can only be const, because they are using a C API underneath, which doesn't enforce const correctness.

There is another reason why not to do: when we are using some kind of read-write locking, then a read lock operation can return a const object, making sure the object doesn't get modified. Because you can have multiple read locks, all of the const functions should be thread-safe from the outside (not modifying the state of the object or using some synchronization internally). Everything that is not thread-safe, shouldn't be const.

What do you think about it? What was your idea, why to make these functions `const`?